language/messages/Reflinks.i18n.php

```diff
@@ -0,0 +1,293 @@
+	'wmflabs-thankyoutest' => 'Obrigado por testar a versão experimental de $1. Favor relatar bugs que você encontrar.',
+	'wmflabs-latestcommit' => 'Commit mais recente: $1',
+	'wmflabs-poweredby' => 'Distribuído por Wikimedia Labs',
+	'wmflabs-testsummary' => 'Formatando $1 referência(s) com a versão teste de $3',
+);
+
+$messages['qqq'] = array(
+	'appname' => '{{Optional|Translate it only if you have a good name that is concise and gives a basic idea of what the tool does in your language.}}',
+	'tagline' => 'A brief explanation of the tool\'s purpose. This is displayed below its name on the main page of the tool. Please keep it concise.',
+	
+	// Headings
+	'heading-fetchfromwiki' => 'The heading of the form of the \'Fetch from wiki\' mode. Used on the main page.',
+	'heading-rawwikitext' => 'The heading of the form of the \'Enter wikitext\' mode. Used on the main page.',
+	'heading-options' => 'The heading of the form sections containing checkboxes of options. Used on the main page.',
+	'heading-citegen' => 'The heading of the form of the citation generator tool. Used on the main page.',
+	'heading-result' => 'The heading of the result page. This is also used as the browser title of the page.',
+	'heading-newwikitext' => 'The heading of the section containing the resulting wikitext. Used on the result page.',
+	'heading-error' => 'The heading of the error page. This is also used as the browser title of the page.',
+	'heading-manual' => 'The heading of the manual page. This is also used as the browser title of the page.',
+	'heading-acknowledgements' => 'The heading of the acknowledgements page. This is also used as the browser title of the page.',
+	'heading-reportbugs' => 'The heading of the bug reporting page. This is also used as the browser title of the page.',
+	
+	// Input placeholders
+	'placeholder-pagename' => 'The placeholder text shown in an input field expecting a name of a wiki page.',
+	'placeholder-url' => 'The placeholder text shown in an input field expecting a URL.',
+	
+	// Labels
+	'label-fixpage' => 'This is used as the button label.',
+	'label-fixwikitext' => 'This is used as the button label.',
+	'label-toggleadv' => 'This is used as the button label.',
+	'label-generate' => 'This is used as the button label.',
+	'label-manual' => 'This is used as the label of a link in the navigation bar.',
+	'label-sourcecode' => 'This is used as the label of a link in the navigation bar.',
+	'label-acknowledgements' => 'This is used as the label of a link in the navigation bar.',
+	'label-reportbugs' => 'This is used as the label of a link in the navigation bar.',
+	'label-switchlang' => 'This is used as the label of a link in the navigation bar.',
+	'label-save' => 'This is used as the button label.',
+	
+	// General messages
+	/*
+		Well, I remember the the translation interface in Launchpad gives an automatic hint about how to deal with the parameters.
+		Maybe the Translate extension can do the same? (Or am I missing something?)
```

Comment:
It already does. Messages are inspected while they type. Omitting parameters yields a warning. The convention is to describe the message and how it is used. Optionally followed by a list of parameters and what they will contain. There's no need to assume the user will remove it, or to explain "run-time" concepts. Translators see those all the time (though using "run-time" probably adds more confusion then it helps avoid).

> Please keep the $1 intact. It\'ll be replaced by a number at run-time.

May be better phrased as:

> Indicates how many references were modified to have citations.
> * $1: The number of fixed references.